e2e/test/iothub/device/FileUploadE2ETests.cs

```diff
@@ -146,16 +145,15 @@ await using (deviceClient)
 
                     try
                     {
-                        // TODO: the HTTP layer handles errors differently and does not produce the right kind of exceptions.
-                        // It should be updated to throw the same kind of exceptions as MQTT and AMQP.
+                        // TODO: Gateway V1 wrongly returns 500 for uploading file with invalid correlation id, while Gateway V2 corrects this behavior.
+                        // Therefore we are handling separately here, and will change this after problem in V1 gets addressed.
                         await deviceClient.CompleteFileUploadAsync(notification).ConfigureAwait(false);
                     }
-                    catch (IotHubClientException ex)
+                    catch (IotHubClientException ex) when (ex.ErrorCode is IotHubClientErrorCode.ServerError)
                     {
                         // Gateway V1 flow
-                        ex.ErrorCode.Should().Be(IotHubClientErrorCode.ServerError);
                     }
-                    catch (ArgumentException ex)
+                    catch (IotHubClientException ex) when (ex.ErrorCode is IotHubClientErrorCode.BadRequest)
                     {
                         // Gateway V2 flow
                         ex.Message.Should().Contain("400006");
```

Comment:
```suggestion
                        ex.ErrorCode.Should().Be(IotHubClientErrorCode.IotHubFormatError);
```

Reply:
Well, that suggestion doesn't make sense given the `when` clause above.

Is there a reason we can't get the error code out?

Reply:
That's reasonable. Sure, I will add that.